### PR TITLE
Uninstallation of Istio

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Add handling for uninstalling Istio when Istio CR is deleted

**Related issue(s)**

- https://github.com/kyma-project/istio/issues/81

### CODE REVIEW

File: operator/internal/reconciliations/istio/installation.go [hunk 1]
```diff
@@ -1,30 +1,99 @@
...
+
+	// We need to evaluate what changed since last reconciliation, to make sure we run Istio reconciliation only if it's necessary
+	istioCRChanges, err := EvaluateIstioCRChanges(istioCR, istioTag)
 	if err != nil {
-		return err
+		ctrl.Log.Error(err, "Error evaluating IstioCR changes")
+		return istioCR, err
+	}
+
+	if !istioCRChanges.MustBeReconciled() {
```
Comment (cnvergence) [inline operator/internal/reconciliations/istio/installation.go:37]: do we need this check at all?

Comment (triffer) [reply operator/internal/reconciliations/istio/installation.go:37]: That's a good point, I think we can remove it.

**Pull request merged**